_ml-commons-plugin/algorithms.md

```diff
@@ -0,0 +1,315 @@
+anomaly_score_threshold | double | The threshold of the anomaly score | 1.0 
+
+#### Fit RCF
+
+Parameter | Type   | Description | Default Value
+:--- |:--- | :--- | :---
+number_of_trees | integer | The number of trees in the forest | 30
+shingle_size | integer | A shingle, or a consecutive sequence of the most recent records | 8
+sample_size | integer | The sample size used by stream samplers in the forest | 256
+output_after | integer | The number of points required by stream samplers before results return | 32
+time_decay | double | The decay factor used by stream samplers in the forest | 0.0001 
+anomaly_rate | double | The anomaly rate | 0.005
+time_field | string | (**Required**) The time filed for RCF to use as time series data | N/A
+date_format | string | The date and time format for the time_field field | "yyyy-MM-ddHH:mm:ss"
+time_zone | string | The time zone for the time_field field | "UTC" 
+
+
+### APIs
+
+* [Train]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#train-model)
+* [Predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#predict)
+* [Train and predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#train-and-predict)
+
+
+### Limitations
+
+For FIT RCF, you can train the model with historical data and store the trained model in your index. The model will be deserialized and predict new data points when using the Predict API. However, the model in the index will not be refreshed with new data, because the model is fixed in time.
+
+## Anomaly Localization 
+
+The Anomaly Localization algorithm finds subset level-information for aggregate data (for example, aggregated over time) that demonstrates the activity of interest, such as spikes, drops, changes, or anomalies. Localization can be applied in different scenarios, such as data exploration or root cause analysis, to expose the contributors driving the activity of interest in the aggregate data.
+
+### Parameters
+
+Parameter | Type   | Description | Default Value
+:--- | :--- | :--- | :---
+index_name | String | The data collection to analyze | N/A
+attribute_field_names | List<String> | The fields for entity keys | N/A
+aggregations | List<AggregationBuilder> | The fields and aggregation for values | N/A
+time_field_name | String | The timestamp field | null
+start_time | Long | The beginning of the time range | 0 
+end_time | Long | The end of the time range | 0
+min_time_interval | Long | The minimum time interval/scale for analysis | 0
+num_outputs | integer | The maximum number of values from localization/slicing | 0
+filter_query | Long | (Optional) Reduces the collection of data for analysis | Optional.empty()
+anomaly_star | QueryBuilder | (Optional) The time after which the data will be analyzed | Optional.empty()
+
+### Example
+
+The following example executes Anomaly Localization against an RCA index. The API responds with 10 aggregations and gives the sum of the contribution and base values per aggregation, every
```

Comment:
It looks like this sentence isn't finished yet?

Reply:
Fixed